src/mscorlib/src/System/Threading/CancellationToken.cs

```diff
@@ -94,13 +74,7 @@ public bool IsCancellationRequested
         /// into a canceled state, meaning that <see cref="IsCancellationRequested"/> will never
         /// return true.
         /// </remarks>
-        public bool CanBeCanceled
-        {
-            get
-            {
-                return m_source != null && m_source.CanBeCanceled;
-            }
-        }
+        public bool CanBeCanceled => _source != null;
```

Comment:
Yes.  Do you see a case where that breaks down?  As far as I can tell, the only way you could get a non-cancelable source previously was with `default(CancellationToken)`/`new CancellationToken()`/`CancellationToken.None`, and all of those (and I believe only those) will have a null CTS.